lib/calculator.rb

```diff
@@ -1,11 +1,18 @@
 require_relative 'printer'
 
 class Calculator
+
   def add(a,b)
-    a + b
+    output(a+b)
   end
 
   def subtract(a,b)
-    a - b
+    output(a-b)
+  end
+
+  def output(number)
+     Printer.new.pretty_print(number)
```

Comment:
This is very tightly coupled - its now impossible to change Printer without affecting Calculator. This is not DI.